include/seqan3/io/concept/stream.hpp

```diff
@@ -0,0 +1,274 @@
+     using traits_type = typename stream_::traits_type;
+     using int_type    = typename stream_::int_type;
+     using off_type    = typename stream_::off_type;
+     using pos_type    = typename stream_::pos_type;
+ };
+
+ template <typename char_t, typename traits_t>
+ struct stream_traits<std::basic_istream<char_t, traits_t>>
+ {
+     using stream_ = std::basic_istream<char_t, traits_t>;
+
+     using char_type   = typename stream_::char_type;
+     using traits_type = typename stream_::traits_type;
+     using int_type    = typename stream_::int_type;
+     using off_type    = typename stream_::off_type;
+     using pos_type    = typename stream_::pos_type;
+ };
+
+ template <typename char_t, typename traits_t>
+ struct stream_traits<std::basic_ifstream<char_t, traits_t>>
+ {
+     using stream_ = std::basic_ifstream<char_t, traits_t>;
+
+     using char_type   = typename stream_::char_type;
+     using traits_type = typename stream_::traits_type;
+     using int_type    = typename stream_::int_type;
+     using off_type    = typename stream_::off_type;
+     using pos_type    = typename stream_::pos_type;
+ };
+
+ template <typename char_t, typename traits_t>
+ struct stream_traits<std::basic_istringstream<char_t, traits_t>>
+ {
+     using stream_ = std::basic_istringstream<char_t, traits_t>;
+
+     using char_type   = typename stream_::char_type;
+     using traits_type = typename stream_::traits_type;
+     using int_type    = typename stream_::int_type;
+     using off_type    = typename stream_::off_type;
+     using pos_type    = typename stream_::pos_type;
+ };
+
+ template <typename char_t, typename traits_t>
+ struct stream_traits<std::basic_ostream<char_t, traits_t>>
+ {
+     using stream_ = std::basic_ostream<char_t, traits_t>;
+
+     using char_type   = typename stream_::char_type;
+     using traits_type = typename stream_::traits_type;
+     using int_type    = typename stream_::int_type;
+     using off_type    = typename stream_::off_type;
+     using pos_type    = typename stream_::pos_type;
+ };
+
+ template <typename char_t, typename traits_t>
+ struct stream_traits<std::basic_ofstream<char_t, traits_t>>
+ {
+     using stream_ = std::basic_ofstream<char_t, traits_t>;
+
+     using char_type   = typename stream_::char_type;
+     using traits_type = typename stream_::traits_type;
+     using int_type    = typename stream_::int_type;
+     using off_type    = typename stream_::off_type;
+     using pos_type    = typename stream_::pos_type;
+ };
+
+ template <typename char_t, typename traits_t>
+ struct stream_traits<std::basic_ostringstream<char_t, traits_t>>
+ {
+     using stream_ = std::basic_ostringstream<char_t, traits_t>;
+     using char_type   = typename stream_::char_type;
+     using traits_type = typename stream_::traits_type;
+     using int_type    = typename stream_::int_type;
+     using off_type    = typename stream_::off_type;
+     using pos_type    = typename stream_::pos_type;
+ };
```

Comment:
lots of code copy'n'paste, you could use one struct template and constrain it to the list of types thereby avoiding copy'n'paste.